src/scan.c

```diff
@@ -210,22 +212,31 @@ const char *token_str(token_t tok)
    return "???";
 }
 
-static const char *get_cond_analysis_identifier(const char *name)
+void conda_id_init()
 {
-   if (strcmp(name, "VHDL_VERSION") == 0)
-      return standard_text(standard());
-   else if (strcmp(name, "TOOL_TYPE") == 0)
-      return "SIMULATION";
-   else if (strcmp(name, "TOOL_VENDOR") == 0)
-      return PACKAGE_URL;
-   else if (strcmp(name, "TOOL_NAME") == 0)
-      return PACKAGE_NAME;
-   else if (strcmp(name, "TOOL_EDITION") == 0)
-      return "";
-   else if (strcmp(name, "TOOL_VERSION") == 0)
-      return PACKAGE_VERSION;
-   else
-      return NULL;
+   cond_idents = shash_new(16);
+
+   conda_id_add("VHDL_VERSION", standard_text(standard()));
+   conda_id_add("TOOL_TYPE",    "SIMULATION");
+   conda_id_add("TOOL_VENDOR",  PACKAGE_URL);
+   conda_id_add("TOOL_NAME",    PACKAGE_NAME);
+   conda_id_add("TOOL_EDITION", "");
+   conda_id_add("TOOL_VERSION", PACKAGE_VERSION);
+
+}
+
+void conda_id_exit()
+{
+   shash_free(cond_idents);
+}
+
+void conda_id_add(const char *name, const char *value)
+{
+   char *existing_val = (char*) shash_get(cond_idents, name);
+   if (existing_val)
+      errorf("conditional analysis directive '%s' already defined (%s)",
+             name, existing_val);
+   shash_put(cond_idents, name, (char*)value);
```

Comment:
```suggestion
   shash_put(cond_idents, name, xstrdup(value));
```

Although it works for all the current usages, it's safer to make a copy of the string. Also don't need a cast as any pointer can be passed to `void *`.